src/lib.rs

```diff
@@ -87,15 +87,13 @@ impl<'a> Iterator for AllocListIterMut<'a> {
         // It's UB to call `.add` on a pointer past its allocation bounds, so we
         // need to check that it's within range before turning it into a pointer
         // and dereferencing it
-        let ulimit = self.alloc_list.start as usize + SIZE_ALLOC_TABLE;
-        let ptr = self.alloc_list.start as usize + (self.index * core::mem::size_of::<Block>());
-
-        self.index += 1;
-
-        if ptr >= ulimit {
+        if self.index * core::mem::size_of::<Block>() >= SIZE_ALLOC_TABLE {
```

Comment:
Random (shallow) comment: would `index` be more idiomatic as `idx`?

Reply:
Fixed